src/modules/jcms_notifications/tests/src/Unit/ArticleCrudServiceTest.php

```diff
@@ -0,0 +1,153 @@
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Tests\UnitTestCase;
+use Drupal\jcms_notifications\ArticleCrudService;
+use \Mockery as m;
+
+/**
+ * Class ArticleCrudServiceTest
+ *
+ * @package Drupal\Tests\jcms_notifications\Unit
+ */
+class ArticleCrudServiceTest extends UnitTestCase {
+
+  /**
+   * @var \Drupal\jcms_notifications\ArticleCrudService
+   *   The class we're testing.
+   */
+  protected $articleCrudService;
+
+  /**
+   * @var \Drupal\jcms_notifications\FetchArticleService
+   */
+  protected $fetchArticleService;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * @var \Drupal\Core\Entity\Query\QueryFactory
+   */
+  protected $entityQuery;
+
+  public function setUp() {
+    $this->fetchArticleService = m::mock('\Drupal\jcms_notifications\FetchArticleService');
+    $this->entityTypeManager = m::mock('\Drupal\Core\Entity\EntityTypeManager');
+    $this->entityQuery = m::mock('\Drupal\Core\Entity\Query\QueryFactory');
+    $container = new ContainerBuilder();
+    $container->set('entity.query', $this->entityQuery);
+    $container->set('entity.manager', $this->entityTypeManager);
+    \Drupal::setContainer($container);
+    $this->articleCrudService = new ArticleCrudService($this->fetchArticleService, $this->entityTypeManager);
+  }
+
+  public function tearDown() {
+    m::close();
+  }
+
+  /**
+   * @test
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::__construct()
+   */
+  public function testCrudArticleService() {
+    $actual = new ArticleCrudService($this->fetchArticleService, $this->entityTypeManager);
+    $this->assertInstanceOf('\Drupal\jcms_notifications\ArticleCrudService', $actual);
+  }
+
+  /**
+   * @test
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::crudArticle()
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::nodeExists()
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::deleteArticle()
+   */
+  public function testCrudArticleDelete() {
+    $this->entityQuery->shouldReceive('get')->once()->andReturn($this->entityQuery);
+    $this->entityQuery->shouldReceive('condition')->once()->andReturn($this->entityQuery);
+    $this->entityQuery->shouldReceive('execute')->once()->andReturn([1234]);
+    $this->entityTypeManager->shouldReceive('getStorage')->once()->andReturn($this->entityTypeManager);
+    $this->entityTypeManager->shouldReceive('delete')->once()->andReturn(NULL);
+    $acutal = $this->articleCrudService->crudArticle([
+      'id' => '123abc',
+      'action' => 'del',
+    ]);
+    $this->assertNull($acutal);
+  }
+
+  /**
+   * @test
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::crudArticle()
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::nodeExists()
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::updateArticle()
+   */
+  public function testCrudArticleUpdate() {
+    $node = m::mock('\Drupal\node\Entity\Node');
```

Comment:
tests that mock ORM calls are going to haunt you in the future as the set of dozens of calls change without the behavior being different. The stable way to test this is at the integration level, actually writing in the database a random article id and checking that is is there. Given that journal-cms has no knowledge of articles before the SQS message coming in, the setup necessary for such a test should be minimal?